Patch windows service options

thxCode opened the pull request:
- Save Windows RKE service options into CRD
- Output the appropriate version images

**Related PR:**
https://github.com/rancher/kontainer-engine/pull/185

**Issue:**
https://github.com/rancher/rancher/issues/22470

kinarashah commented on pkg/controllers/management/kontainerdrivermetadata/data_getter.go:
```diff
@@ -105,47 +105,31 @@ func GetK8sVersionInfo(rancherVersion string, rkeSysImages map[string]v3.RKESyst
-				return k8sSvcOption, err
-			}
+		// detect the svc options existing or not
+		if _, exist := svcOptions[majorVersion]; !exist {
```
we should first search for `svcOptions[k8sVersion]`, if it exists, store it for that version. If not, search for `majorVersion`. This is because we're supporting service options for both

thxCode replied:
Updated

kinarashah commented on pkg/controllers/management/kontainerdrivermetadata/data_getter.go:
```diff
@@ -105,47 +105,31 @@ func GetK8sVersionInfo(rancherVersion string, rkeSysImages map[string]v3.RKESyst
+
+		rkeSysImages := rkeSysImages[k8sVersion]
+		switch osType {
+		case Windows:
```
for windows we should also add a check for k8sVersion's. we should continue if `majorVersion < v1.14`

thxCode replied:
Yes, I use the existing checking of `svcOptions[majorVersion]` to avoid those unsupported versions.

kinarashah commented on pkg/image/export/main.go:
```diff
@@ -88,6 +86,9 @@ func run(images ...string) error {
 		targetImages = append(targetImages, image.Mirror(i))
 	}
 
+	k8sVersionToWindowsSystemImages, _ := kd.GetK8sVersionInfo(rancherVersion,
```
is there some way to combine this for both linux and windows?  Actual logic of calculating k8s versions is happening twice. This function gets called only by `pkg/image/export/main.go` so feel free to make changes as you like :-)

thxCode replied:
Updated

alena1108 commented:
@thxCode rke pr has been merged, can you rebase this pr with the official vendor? and squash the changes into the code and vendor commits

thxCode commented:
@alena1108 , I have updated the rke vendor and squashed other commits, but this PR needs https://github.com/rancher/kontainer-engine/pull/185 support, I will test it with the steps described by @kinarashah , and then ping you later.

thxCode commented:
@kinarashah , @alena1108 , updated `rancher/kontianer-engine` vendor.

loganhz commented:
Close in favor of https://github.com/rancher/rancher/pull/22738
